Jackd support for the GUI version

alexolivan opened the pull request:
Not sure if I'm doing this right... I'm not used to PR.

But the code changes have been tested for several months now... and I've had no issues at all. In reality there's nothing very interesting on the changes:

Basically, I've added the ability to (optionally) pass the glassplayer CLI params required to connect to jack, to the glassplayer-gui binary call, so it can in turn pass them when spawning the player.
When passing the jack client name, the resulting window title prepends the jack client name to the tittle, so it is easier to distinguish between multiple glassplayer-gui instances running as several, independent, jack clients.
I also changed the 'Metadata tittle' window title, according to the comment, because it was confusing to me... just cosmetic
Probably not the best coding practices, and I guess any changes should be documented on the man pages... but the functionality is very very useful to me. I missed very very much jackd on glassplayer-gui, while glassplayer, glasscoder and, overall, the rivendell ecosystem is fully jackd-friendly.

Best regards.

alexolivan committed:
Added jackd support.

This is essentially adding passthrough to glassplayer of any of the 3  jackd-related arguments.

alexolivan committed:
Clean and issue a PR

Clean the source. Try to submit a Pull Request

ElvishArtisan commented:
Some very sensible additions! Might I suggest a few tweaks:

1. Variable names. Avoid 'camel-case' variable names, such as 'jackArgs'. A better choice would be 'jack_args'. That is the style used throughout the existing codebase.
2. Please add a ```ChangeLog``` entry describing your changes.
3. Document your changes in the ```glassplayergui(1)``` man page. See the 'docs/glassplayergui.xml' file. It's pretty simple once you see an example of how it works.

Details and rationale for all of the above can be found in the 'CODINGSTYLE' file the Rivendell repo (in fact, I should put a copy of that document into each of my repos.)

Edit: Turns out that I already did this. (Duh! I really should finish my coffee before processing these.) So, see 'CODINGSTYLE' in this repo (especially the 'PULL REQUEST CHECKLIST' section near the end).

alexolivan commented:
Hi!

Understood ... will read the CODINGSTYLE and try again :-P

Best regards!

alexolivan committed:
Add man page documentation.

Add documentaation of supported options to be passed through to grlassplayer. Since any other option is rejeceted by design, Just copy/paste glassplayer man relevant sections and state no other parameters will work other than the ones refered to ALSA.

alexolivan committed:
Refactor code to CODINGSTYLE guidelines

Change variable names to follow origin's CODINGSTYLE guidelines

alexolivan committed:
Fix xml errors and indentation on MANpage changes

alexolivan committed:
Upadated the changelog

Updated the change log to reflect code changes.

alexolivan committed:
Fix bad style and tabs on Changelog

alexolivan committed:
Fix missing variable refactor to CODINGSTYLE

alexolivan committed:
Fix build error

alexolivan commented:
OK... I thing I did it. I refactored the code, added the XML based on the existing for glassplayer, and updated the changelog.
I had to solve a pair of build errors (bad xml, and then I miss a refactored var) and the sources build...  but now I'm seeing there're code conflicts and that's scaring to me...

I tested the built binary and apparently it worked as it did before the refactors/changes.

Best regards.

UPDATE: Tried to fix the Changelog conflict. As usually I didn't right and the changelog in my working branch was outdated... The thing is that I have fixed it, so all changes are now in correct chronological order, with my lines sitting at the bottom, just after the last ones in the original file, but the conflict persists.

alexolivan committed:
Try to fIx Changelog conflict

ElvishArtisan commented:
No worries about conflicts in 'ChangeLog'. Those are routine in PRs. I fix those when I merge it.

ElvishArtisan commented:
Merged to 'master'. Thank you!

alexolivan commented:
HI Fred!
Thanks for the merge.

Please consider releasing some tarballs of GlassPlayer and GlassCoder with the current debian-ready code, even as new beta 
 version or something ... with the current release tarballs, from 2019/20, I can't get debian packages 'the right way'.
I'm currently packaging both 'release', and niglhtly-build 'snapshot', versions of Rivendell packages, but I can just get nightly-build 'snapshots' from GlassPlayer/Coder, no 'release' packages still for the 'extras' component of my repo.

Anyhow.
Thank you so much.
